packages/typespec-vscode/test/manual/typespec-extension-basic-features-test-plan.md

```diff
@@ -0,0 +1,158 @@
+# TypeSpec Extension Basic Features Test Scenario
+
+The TypeSpec for VS Code extension provides TypeSpec language support for VS Code.
+
+## Features
+
+- [Syntax highlighting](#for-syntax-highlighting)
+- [Live diagnostic reporting](#for-live-diagnostic-reporting)
+- [Code completion](#for-code-completion)
+- [Code folding](#for-code-folding)
+- [Formatting](#for-formatting)
+- [Hover info](#for-hover-info)
+- [Rename refactoring](#for-rename-refactoring)
+- [Go to definition](#for-go-to-definition)
+
+## Test Environment
+
+- OS : Windows or Linux
+- Editor: Visual Studio Code
+
+> Note: The extension should support all test cases in VS Code for Windows and Linux. Mac support is a stretch goal for Selenium semester.
+
+## Test Steps
+
+### Step 1: Open a TypeSpec project and view a .tsp file.
+
+**Validate:** The code color is gray.
+
+![alt text](./images/TypeSpecForvscodeFeatures_BeforeInstallExtension.png)
+
+### Step 2: Install the TypeSpec for VS Code extension.
+
+_Option 1_. Install using .vsix file:
+`Extension` -> `…` -> `Install form VSIX...`
+
+![alt text](./images/InstallTypespec_VSIX.png)
+
+Find the .vsix file you want to install locally.
+
+![alt text](./images/InstallTypeSpec_SelectVSIXFileTest.png)
+
+_Option 2_. Install typespec with vscode extension marketplace:
+`Extension` -> input `TypeSpec for VS Code` -> `Install`
+
+![alt text](./images/InstallTypespec_ExtensionMarketplaceTest01.png)
+
+### Step 3: Install TypeSpec Compiler CLI
+
+[Install TypeSpec Compiler CLI](https://typespec.io/docs/) command: `"npm install -g @typespec/compiler"`.
+
+### Step 4: Verify Basic Features
+
+- #### For Syntax highlighting:
+
+  View the .tsp file again, check the display of keywords, types, comments, and so on in the file.
+
+  **Validate:** Keywords, types, comments, etc. should be highlighted in different colors.
+
+  ![alt text](./images/TypeSpecForvscodeFeatures_SyntaxHighlighting.png)
+
+- #### For Live diagnostic reporting:
+
+  1. Introducing errors in the file, such as declaring an undefined type.
+
+     ![alt text](./images/TypeSpecForvscodeFeatures_DeclaringUndefinedTypes.png)
+
+  2. View error message.
+
+     **Validate:** A real-time diagnostic error message should be displayed.
+
+     ![alt text](./images/TypeSpecForvscodeFeatures_ErroeMessages.png)
+
+- #### For Code completion:
+
+  Start typing keywords or press `ctrl + space` in the file.
+
+  **Validate:** Whether to automatically pop up a completion prompt to list possible types, attributes, and operations.
+
+  - For `.tsp` file:
```

Comment:
the screenshot shown is normal completion from vscode, not from typespec. please verify the completion from typespec model or operation.